stan/math/opencl/kernels/tri_multiply.hpp

```diff
@@ -0,0 +1,155 @@
+#ifndef STAN_MATH_OPENCL_KERNELS_TRI_MULTIPLY_HPP
+#define STAN_MATH_OPENCL_KERNELS_TRI_MULTIPLY_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_cl.hpp>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+// \cond
+static const char* tri_multiply_kernel_code = STRINGIFY(
+    // \endcond
+    /**
+     * OpenCL Matrix multiplication of a triangular and rectangular matrix
+     *
+     * @param[in] A the lower/upper triangular matrix in matrix multiplication
+     * @param[in] B the right matrix in matrix multiplication
+     * @param[out] C the output matrix
+     * @param[in] M Number of rows for matrix A
+     * @param[in] N Number of rows for matrix B
```

Comment:
cols?

Reply:
Ah, have to fix this in the regular multiply also.